src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -376,30 +374,26 @@ public ReadResponse createResponse(UnfilteredPartitionIterator iterator)
 
     long indexSerializedSize(int version)
     {
-        return null != index
-             ? IndexMetadata.serializer.serializedSize(index, version)
+        return null != indexQueryPlan
+             ? IndexMetadata.serializer.serializedSize(indexQueryPlan.getFirst().getIndexMetadata(), version)
              : 0;
     }
 
     public Index getIndex(ColumnFamilyStore cfs)
```

Comment:
`cfs` is no longer needed so could be removed.

Reply:
I will remove it and move `Index#postProcessorFor` to `QueryPlan`.